### PR TITLE
AWS-SDK: Fixing broken exports

### PR DESCRIPTION
**Pull request by engelhartrueben**
# Fixes # (issue)
Fixes broken exports.

## Description
During the upgrade to Node 20, `aws-sdk` had to be upgraded to V3. This was simple in nature, considering a code-mod existed. However, our specific way of initiating an S3 bucket was not covered by this code-mod, and subsequently failed. Theoretically, our implementation (still in V2) was still supported, but clearly not.

This new implementation uses `aws-sdk` V3 api calls to create the S3 client and bucket. Additionally, we can set the region for the bucket, which is called from the environment variables. 

This PR:
- Checks if an AWS bucket exists, if not creates one of the specificed name found in `AWS_S3_BUCKET_NAME=`
- Brings v3 AWS-SDK update to export logic
- Only shows UI related to the type of export:
Local Export: 
image
AWS Export: 
image
(__NOTE__) this UI is fixed with #2423  
- Places a hard check on if an email (SMTP or MailGun) is set up for exports
- Adjusts the expiration for cached `campaignExportUrl` to match AWS expiration (1 day)

Tested on:
| Mode | AWS Bucket and Download| Local download | 
| --- | --- | --- |
| production | ✔️ | ❌*|
| development | ✔️|✔️|

_*Local download is not supported in production._
# Checklist:

- [x] I have manually tested my changes on desktop and mobile
- [x] The test suite passes locally with my changes
- [x] If my change is a UI change, I have attached a screenshot to the description section of this pull request
- [x] [My change is 300 lines of code or less](https://github.com/StateVoicesNational/Spoke/blob/main/CONTRIBUTING.md#submitting-your-pull-request), or has a documented reason in the description why it’s longer
- [x] I have made any necessary changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] My PR is labeled [WIP] if it is in progress

### CODE REVIEW

File: src/containers/AdminCampaignStats.jsx [hunk 1]
```diff
@@ -424,9 +424,10 @@ class AdminCampaignStats extends React.Component {
           message={
             <span>
               Export started -
-              {this.props.organizationData &&
-                this.props.organizationData.emailEnabled &&
-                " we'll e-mail you when it's done. "}
+              {(this.props.organizationData &&
+                this.props.organizationData.emailEnabled) ?
+                " we'll e-mail you when it's done. " :
+                " check your Spoke directory. "}
```
Comment (mau11) [inline src/containers/AdminCampaignStats.jsx]: I just tried this locally, if we have AWS S3 set up, even if email isn't enabled, the data will save to S3, not in the spoke directory

File: src/workers/jobs.js [hunk 2]
```diff
@@ -861,46 +861,74 @@ export async function exportCampaign(job) {
...
-        // The transformation for params is not implemented.
-        // Refer to UPGRADING.md on aws-sdk-js-v3 for changes needed.
-        // Please create/upvote feature request on aws-sdk-js-codemod for params.
-        params: { Bucket: process.env.AWS_S3_BUCKET_NAME }
+      const client = new S3Client({
+        region: process.env.AWS_REGION
       });
+      const bucketName = process.env.AWS_S3_BUCKET_NAME;
+      const command = new CreateBucketCommand({ Bucket : bucketName });
+
```
Comment (mau11) [inline src/workers/jobs.js:875]: I don't think we want to create a new bucket each time we export data, it's likely someone will already have bucket created for the data and want to upload directly to it

we can add a check, if the bucket exists, save the exported data, if it doesn't, create a new one and save

File: src/workers/jobs.js [hunk 3]
```diff
@@ -865,14 +872,46 @@ export async function exportCampaign(job) {
...
+      try {
+        // Check if the S3 bucket already exists
+        const verifyBucketCommand = new HeadBucketCommand({
+          Bucket: bucketName
+        });
+        await client.send(verifyBucketCommand);
+
+        console.log(`S3 bucket "${bucketName}" already exists.`);
+      } catch (error) {
+        if (error.name === "NotFound") {
```
Comment (engelhartrueben) [inline src/workers/jobs.js:885]: `if (error.code === "NoSuchBucket")`

[Source](https://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html)

Comment (mau11) [reply src/workers/jobs.js:885]: When the bucket is not found, it doesn't look like we get this error code, only the error name

File: src/containers/AdminCampaignStats.jsx (hunk 1 above)
Comment (mau11) [reply src/containers/AdminCampaignStats.jsx]: disregard my previous comment, the last change to this check fixed the issue I was seeing

**Comment by engelhartrueben** [comment]
@mau11 this is set up for `main`, but can rebase to `stage-main-14.1` upon request!

**Review by mau11 — APPROVED** [review]
🎉

**Pull request merged**